scripts/stitching/stitching.py

```diff
@@ -32,203 +35,291 @@ def combine_entity_by_name(roi_fov_paths, cols, rows, enforce_square=False, marg
-                new_im_metadata['mass_gain'] = metadata['mass_gain']
-                new_im_metadata['mass_offset'] = metadata['mass_offset']
-                new_im_metadata['x_coord'] = int(np.round(um_min_x))
-                new_im_metadata['y_coord'] = int(np.round(um_min_y))
-                new_im_metadata['tissue'] = metadata['raw_description']['fov']['section']['tissue'] \
-                    and metadata['raw_description']['fov']['section']['tissue']['id']
-                new_im_metadata['section'] = metadata['raw_description']['fov']['section']['id']
-                new_im_metadata['aperture'] = metadata['aperture']
-                new_im_metadata['imaging_preset'] = metadata['imaging_preset']
-                new_im_metadata['lens1_voltage'] = metadata['lens1_voltage']
-
-                if not full_id:
-                    for t in range(MAX_TRIES):
-                        try:
                             _ = mr.post('/images/', json=new_im_metadata)
-                            break
-                        except:
-                            if t < MAX_TRIES-1:
-                                time.sleep(0.50)
-                            else:
-                                _ = mr.post('/images/', json=new_im_metadata)
-                else:
-                    for t in range(MAX_TRIES):
-                        try:
-                            _ = mr.put(f'/images/{full_id}', json=new_im_metadata)
-                            break
-                        except:
-                            if t < MAX_TRIES-1:
-                                time.sleep(0.50)
-                            else:
-                                _ = mr.put(f'/images/{full_id}', json=new_im_metadata)
-
+            else:
                 for t in range(MAX_TRIES):
                     try:
-                        mr.upload_mibitiff(out_path, run_id=run_id)
+                        mr.put(f'/images/{full_id}', json=new_im_metadata)
                         break
                     except:
                         if t < MAX_TRIES-1:
                             time.sleep(0.50)
                         else:
-                            mr.upload_mibitiff(out_path, run_id=run_id)
-                
-                print(f"Stitched MIBItiff, {out_path}, uploaded to MIBItracker.")
-                        
+                            mr.put(
+                                f'/images/{full_id}', json=new_im_metadata)
+
+            for t in range(MAX_TRIES):
+                try:
+                    mr.upload_mibitiff(out_path, run_id=run_id)
+                    break
+                except:
+                    if t < MAX_TRIES-1:
+                        time.sleep(0.50)
+                    else:
+                        mr.upload_mibitiff(out_path, run_id=run_id)
+            
+            print(
+                f"Stitched MIBItiff, {out_path}, uploaded to MIBItracker.")
+
+def parse_args(args):
```

Comment:
Instead of specifying variables in the script itself, let the users pass them as arguments when running it